src/System.Private.ServiceModel/tests/Scenarios/Extensibility/WebSockets/WebSocketTests.cs

```diff
@@ -134,11 +139,18 @@ public static void WebSocket_Http_Duplex_BinaryStreamed()
 
             client.StopPushingStream();
             // Waiting on ReceiveStreamCompleted from the ClientReceiver.
-            clientReceiver.ReceiveStreamCompleted.WaitOne();
+            Assert.True(clientReceiver.ReceiveStreamCompleted.WaitOne(ScenarioTestHelpers.TestTimeout),
+                "Test case timeout was reached while waiting for the stream response from the Service to be completed.");
             clientReceiver.ReceiveStreamCompleted.Reset();
 
+            // Getting results from server via callback.
+            client.GetLog();
+            Assert.True(clientReceiver.LogReceived.WaitOne(ScenarioTestHelpers.TestTimeout),
```

Comment:
Question only: you use `ScenarioTestHelpers.TestTimeout` all the time here, but it's not really reflective of the total amount of time spent in the test. Is it necessary to use the TimeoutHelper to reduce the blocking time during each WaitOne call? 

(I don't know that it is, so it's ok with me as long as it's been thought about)